v2/index/index_test.go

```diff
@@ -0,0 +1,177 @@
+package index
+
+import (
+	"bytes"
+	"io"
+	"os"
+	"path"
+	"testing"
+
+	blocks "github.com/ipfs/go-block-format"
+	"github.com/ipld/go-car/v2/internal/carv1"
+	"github.com/ipld/go-car/v2/internal/carv1/util"
+	"github.com/multiformats/go-multicodec"
+	"github.com/multiformats/go-varint"
+	"github.com/stretchr/testify/require"
+)
+
+func TestNew(t *testing.T) {
+	tests := []struct {
+		name    string
+		codec   multicodec.Code
+		want    Index
+		wantErr bool
+	}{
+		{
+			"CarSortedIndexCodecIsConstructed",
+			multicodec.CarIndexSorted,
+			newSorted(),
+			false,
+		},
+		{
+			"ValidMultiCodecButUnknwonToIndexIsError",
+			multicodec.Cidv1,
+			nil,
+			true,
+		},
+		{
+			"IndexSingleSortedMultiCodecIsError",
+			multicodec.Code(indexSingleSorted),
+			nil,
+			true,
+		},
+		{
+			"IndexHashedMultiCodecIsError",
+			multicodec.Code(indexHashed),
+			nil,
+			true,
+		},
+		{
+			"IndexGobHashedMultiCodecIsError",
+			multicodec.Code(indexGobHashed),
+			nil,
+			true,
```

Comment:
please use explicit keys for these; otherwise I just see `nil, true,` and I have no clue what they mean without scrolling back to the top.

not a huge problem here given the few tests, but it's good to be consistently specific for table-driven tests, for clarity - and because we might add more tests later :)